packages/transformers/babel/src/env.js

```diff
@@ -1,52 +1,44 @@
 // @flow
+import semver from 'semver';
 
-import type {Engines, MutableAsset} from '@parcel/types';
+import type {IConfig, Engines} from '@parcel/types';
+import getTargetEngines from './getTargetEngines';
 
 import presetEnv from '@babel/preset-env';
-import semver from 'semver';
-import getTargetEngines from './getTargetEngines';
 
 /**
  * Generates a @babel/preset-env config for an asset.
  * This is done by finding the source module's target engines, and the app's
  * target engines, and doing a diff to include only the necessary plugins.
  */
-export default async function getEnvConfig(
-  asset: MutableAsset,
-  isSourceModule: boolean
-) {
+export default async function getEnvOptions(config: IConfig) {
   // Load the target engines for the app and generate a @babel/preset-env config
-  let targetEngines = asset.env.engines;
-  let targetEnv = await getEnvPlugins(targetEngines, true);
-  if (!targetEnv) {
-    return null;
-  }
+  let targetEngines = config.env.engines;
+  let envOptions = await getPresetOptions(targetEngines, true);
 
   // If this is the app module, the source and target will be the same, so just compile everything.
   // Otherwise, load the source engines and generate a babel-present-env config.
-  if (!isSourceModule) {
-    let sourceEngines = await getTargetEngines(asset);
-    let sourceEnv = (await getEnvPlugins(sourceEngines, false)) || targetEnv;
+  if (!(await config.isSource())) {
+    let sourceEngines = await getTargetEngines(config);
+    if (!sourceEngines) return null;
 
-    // Do a diff of the returned plugins. We only need to process the remaining plugins to get to the app target.
```

Comment:
I'm not sure how relevant this comment is anymore. I still diff the plugins to see if we need to transform with the env preset, but the diffed plugins are all just functions right? I have no way of knowing *which* plugins need to run in a cacheable and rehydratable way. Unfortunately this means that more plugins than necessary will run over these linked packages, but I feel like it's a problem that could possibly be solved in the preset itself. If the preset accepted another option like `startingTarget`, that would be perfect. I thought I added a comment suggesting this, but maybe not because I can't find it now. I suppose it would still be helpful to leave a comment describing why we diffing the plugins though.

Reply:
Added a more accurate comment